### PR TITLE
15_2_13 Border value type validation

### PR DESCRIPTION
**Pull request by sergeroshe**

### CODE REVIEW

File: 15_mini_project_work/2_num_guessing_game.py [hunk 1]
```diff
@@ -42,11 +46,28 @@ def is_valid(input_string, guess_left_border, guess_right_border):
...
+
 def guessing_game_run():
     is_guess_wrong = True
     guess_count = 0
-    guess_left_border = int(input(LEFT_BORDER_PROMPT_MESSAGE))
-    guess_right_border = int(input(RIGHT_BORDER_PROMPT))
+    guess_left_border = input(LEFT_BORDER_PROMPT_MESSAGE)
+    guess_left_border = get_num_input(guess_left_border)
+
+    guess_right_border = input(RIGHT_BORDER_PROMPT)
```
Comment (YanumVlad) [inline 15_mini_project_work/2_num_guessing_game.py:69]: Move the initial input to the get_nym_input()
it is logical: you are asking to get input until done

**Review by YanumVlad — CHANGES_REQUESTED** [review]

File: 15_mini_project_work/2_num_guessing_game.py (hunk 1 above)
Comment (sergeroshe) [reply 15_mini_project_work/2_num_guessing_game.py:69]: Initial input - you mean guess not border right?

Comment (sergeroshe) [reply 15_mini_project_work/2_num_guessing_game.py]: ok

File: 15_mini_project_work/2_num_guessing_game.py [hunk 2]
```diff
@@ -42,11 +46,28 @@ def is_valid(input_string, guess_left_border, guess_right_border):
...
 
+#  returns integer input, asks untill get valid input
+def get_num_input(prompt):
+    is_string_num = False
+    input_string = input(prompt)
+    while not is_string_num:
+        if input_string[0] == MINUS_CHAR and input_string[1:].isdigit() or input_string.isdigit():
+            is_string_num = True
+        else:
+            print(TYPE_ERROR_MESSAGE)
```
Comment (YanumVlad) [inline 15_mini_project_work/2_num_guessing_game.py:57]: Let's set error message as input param

File: 15_mini_project_work/2_num_guessing_game.py [hunk 3]
```diff
@@ -42,11 +46,28 @@ def is_valid(input_string, guess_left_border, guess_right_border):
     return result
 
 
+#  returns integer input, asks untill get valid input
+def get_num_input(prompt):
+    is_string_num = False
+    input_string = input(prompt)
+    while not is_string_num:
+        if input_string[0] == MINUS_CHAR and input_string[1:].isdigit() or input_string.isdigit():
```
Comment (YanumVlad) [inline 15_mini_project_work/2_num_guessing_game.py:54]: A bit overhead, let's hard-code it in THIS particular case

**Review by YanumVlad — CHANGES_REQUESTED** [review]

**Pull request merged**